trpcultivate_genotypes/tests/src/Functional/GenotypesLoader/GenotypesLoaderBasePluginTest.php

```diff
@@ -154,5 +157,100 @@ public function testGenotypesLoaderPluginBase(){
     // Get sample filepath
     $grabbed_sample_file_path = $plugin->getSampleFilepath();
     $this->assertEquals($sample_file_path, $grabbed_sample_file_path, "The sample filepath grabbed by the getter method does not match.");
+
+    /****************************************************************************
+     *  TESTS for method getRecordPkey()
+     ****************************************************************************/
+    // Get the primary key of the organism we created using getRecordPkey()
+    $organism_id = $plugin->getRecordPkey('Organism', 'organism', 0, array(
+      'genus' => $genus,
+      'species' => $species
+    ));
+    $this->assertEquals($organism_id, $grabbed_organism_id, "The organism Id grabbed by getRecordPkey() does not match.");
+
+    // Try to select a record using an invalid mode
+    $exception_caught = FALSE;
+    try {
+      $organism_id = $plugin->getRecordPkey('Organism', 'organism', 5, array(
+        'genus' => $genus,
+        'species' => $species
+      ));
+    }
+    catch ( \Exception $e ) { 
+      $exception_caught = TRUE;
+    }
+    $this->assertTrue($exception_caught, "Did not catch exception for using an inavlid mode (5).");
+
+    // Try to insert a record that already exists and catch the expected exception
+    $exception_caught = FALSE;
+    try {
+      $organism_id = $plugin->getRecordPkey('Organism', 'organism', 1, array(
+        'genus' => $genus,
+        'species' => $species
+      ));
+    } 
+    catch ( \Exception $e ) { 
+      $exception_caught = TRUE;
+    }
+    $this->assertTrue($exception_caught, "Did not catch exception for trying to insert a duplicate.");
+
+    // Enter a duplicate record to trigger an exception
+    $organism_id = $connection->insert('1:organism')
+    ->fields(['genus', 'species'])
```

Comment:
Why is there a `fields` added to an insert query 🤔

Reply:
All of my inserts are using fields. 😬 But I'm not sure why it's wrong... I've been following the Drupal documentation here: https://www.drupal.org/docs/drupal-apis/database-api/insert-queries#compact-form 
I guess I can see that "fields" may not be necessary at all if the name of the field is also included in "values". What about in [this case](https://github.com/TripalCultivate/TripalCultivate-Genetics/blob/336b7ba8d341204ca2c084dcfde51563a4c06a0f/trpcultivate_genotypes/src/GenotypesLoader/GenotypesLoaderPluginBase.php#LL202C2-L205C1)? Can I use "values" instead of "fields" altogether?

      // Insert all of our values
      $result = $this->connection->insert('1:' . $table)
        ->fields($values)
        ->execute();